tests/test_job.py

```diff
@@ -51,7 +51,7 @@ def test_get_job_details_with_no_job_id(self, id, mock_client):
         ['unauthorized', 'job-not-found']))
     def test_get_job_details_with_error_response(self, error, mock_client, make_mock_response):
         status = error.get('status')
-        response = make_mock_response(url=JOB_ID_URL, status=status, json_data=error)
+        response = make_mock_response(url=JOB_ID_URL, status=status, json_data=error, text=status)
```

Comment:
To make sure we handle the message text properly I would set text to something else along with the match parameter on `pytest.raises`

Reply:
Please do the same for all the similar tests